interp/src/values.rs

```diff
@@ -161,12 +161,64 @@ impl Value {
     pub fn as_u64(&self) -> u64 {
         let mut val: u64 = 0;
         for (index, bit) in self.vec.iter().by_ref().enumerate() {
-            val += u64::pow(2, (index as usize).try_into().unwrap())
-                * (*bit as u64);
+            if *bit {
+                //protects against panic in case of # less than u64::max in
+                // value of width greater than 64
+                val += u64::pow(2, (index as usize).try_into().unwrap())
+                    * (*bit as u64);
```

Comment:
I know this is old code, but just checking: would `val |= 1 << index` work as opposed to needing `pow` and stuff?

Reply:
It should, might simplify things slightly, though I don't know what the default behavior of a shift on runtime overflow is. My best guess would be assuming it behaves like addition (panic in debug, wrap in prod)

Reply:
I would never have thought of replacing this w/ `|=`, cool!

Reply:
FWIW, it looks like `<<` and `>>` get masked to the bit width, so "going off the end" would turn into a no-op here. https://rust-lang.github.io/rfcs/0560-integer-overflow.html#arithmetic-operations-with-error-conditions